libs/community/langchain_community/utilities/notdiamond.py

```diff
@@ -0,0 +1,306 @@
+                    llm_config = LLMConfig.from_string(llm_config)
+                if llm_config.provider not in _LANGCHAIN_PROVIDERS:
+                    raise ValueError(
+                        f"Requested provider in {llm_config} supported by Not Diamond "
+                        "but not langchain.chat_models.base.init_chat_model. Please "
+                        "remove it from your llm_configs."
+                    )
+
+        try:
+            nd_version = metadata.version("notdiamond")
+        except (AttributeError, metadata.PackageNotFoundError):
+            nd_version = "none"
+
+        nd_client.user_agent = f"langchain-community/{nd_version}"
+
+        self.client = nd_client
+        self.api_key = nd_client.api_key
+        self.llm_configs = nd_client.llm_configs
+        self.nd_kwargs = nd_kwargs or dict()
+
+    def _model_select(self, input: LanguageModelInput) -> str:
+        messages = _convert_input_to_message_dicts(input)
+        _, provider = self.client.chat.completions.model_select(
+            messages=messages, **self.nd_kwargs
+        )
+        provider_str = _nd_provider_to_langchain_provider(str(provider))
+        return provider_str
+
+    async def _amodel_select(self, input: LanguageModelInput) -> str:
+        messages = _convert_input_to_message_dicts(input)
+        _, provider = await self.client.chat.completions.amodel_select(
+            messages=messages, **self.nd_kwargs
+        )
+        provider_str = _nd_provider_to_langchain_provider(str(provider))
+        return provider_str
+
+    def stream(
+        self,
+        input: LanguageModelInput,
+        config: Optional[RunnableConfig] = None,
+        **kwargs: Optional[Any],
+    ) -> Iterator[str]:
+        yield self._model_select(input)
+
+    def invoke(
+        self, input: LanguageModelInput, config: Optional[RunnableConfig] = None
+    ) -> str:
+        return self._model_select(input)
+
+    def batch(
+        self,
+        inputs: Sequence[LanguageModelInput],
+        config: Optional[Union[RunnableConfig, List[RunnableConfig]]] = None,
+        **kwargs: Optional[Any],
+    ) -> List[str]:
+        return [self._model_select(input) for input in inputs]
+
+    async def astream(
+        self,
+        input: LanguageModelInput,
+        config: Optional[RunnableConfig] = None,
+        **kwargs: Optional[Any],
+    ) -> AsyncIterator[str]:
+        yield await self._amodel_select(input)
+
+    async def ainvoke(
+        self,
+        input: LanguageModelInput,
+        config: Optional[RunnableConfig] = None,
+        **kwargs: Optional[Any],
+    ) -> str:
+        return await self._amodel_select(input)
+
+    async def abatch(
+        self,
+        inputs: Sequence[LanguageModelInput],
+        config: Optional[Union[RunnableConfig, List[RunnableConfig]]] = None,
+        **kwargs: Optional[Any],
+    ) -> List[str]:
+        return [await self._amodel_select(input) for input in inputs]
```

Comment:
```suggestion
    def invoke(
        self, input: LanguageModelInput, config: Optional[RunnableConfig] = None
    ) -> str:
        return self._model_select(input)

    async def ainvoke(
        self,
        input: LanguageModelInput,
        config: Optional[RunnableConfig] = None,
        **kwargs: Optional[Any],
    ) -> str:
        return await self._amodel_select(input)
```

rely on the base implementations